contracts/vaults/STETHVault/base/RibbonVault.sol

```diff
@@ -0,0 +1,661 @@
+     ***********************************************/
+
+    /**
+     * @notice Sets the new fee recipient
+     * @param newFeeRecipient is the address of the new fee recipient
+     */
+    function setFeeRecipient(address newFeeRecipient) external onlyOwner {
+        require(newFeeRecipient != address(0), "!newFeeRecipient");
+        feeRecipient = newFeeRecipient;
+    }
+
+    /**
+     * @notice Sets the management fee for the vault
+     * @param newManagementFee is the management fee (6 decimals). ex: 2 * 10 ** 6 = 2%
+     */
+    function setManagementFee(uint256 newManagementFee) external onlyOwner {
+        require(newManagementFee < 100 * 10**6, "Invalid management fee");
+        // We are dividing annualized management fee by num weeks in a year
+        managementFee = newManagementFee.mul(10**6).div(WEEKS_PER_YEAR);
+    }
+
+    /**
+     * @notice Sets the performance fee for the vault
+     * @param newPerformanceFee is the performance fee (6 decimals). ex: 20 * 10 ** 6 = 20%
+     */
+    function setPerformanceFee(uint256 newPerformanceFee) external onlyOwner {
+        require(newPerformanceFee < 100 * 10**6, "Invalid performance fee");
+        performanceFee = newPerformanceFee;
+    }
+
+    /**
+     * @notice Sets a new cap for deposits
+     * @param newCap is the new cap for deposits
+     */
+    function setCap(uint104 newCap) external onlyOwner {
+        require(newCap > 0, "!newCap");
+        vaultParams.cap = newCap;
+    }
+
+    /************************************************
+     *  DEPOSIT & WITHDRAWALS
+     ***********************************************/
+
+    /**
+     * @notice Deposits ETH into the contract and mint vault shares.
+     */
+    function depositETH() external payable nonReentrant {
+        require(msg.value > 0, "!value");
+
+        _depositFor(msg.value, msg.sender, true);
+    }
+
+    /**
+     * @notice Deposits the `collateralAsset` into the contract and mint vault shares.
+     * @param amount is the amount of `collateralAsset` to deposit
+     */
+    function depositYieldToken(uint256 amount) external nonReentrant {
+        require(amount > 0, "!amount");
+
+        _depositFor(
+            collateralToken.getStETHByWstETH(amount),
+            msg.sender,
+            false
+        );
+
+        IERC20(collateralToken.stETH()).safeTransferFrom(
+            msg.sender,
+            address(this),
+            amount
+        );
+    }
+
+    /**
+     * @notice Mints the vault shares to the creditor
+     * @param amount is the amount of `asset` deposited
+     * @param creditor is the address to receieve the deposit
+     * @param isETH is whether this is a depositETH call
+     */
+    function _depositFor(
+        uint256 amount,
+        address creditor,
+        bool isETH
+    ) private {
+        uint256 currentRound = vaultState.round;
+        uint256 totalWithDepositedAmount =
+            isETH ? totalBalance() : totalBalance().add(amount);
```

Comment:
added an isETH param because totalBalance() for eth deposits will have already included the deposit (msg.value not transerFrom), so we double count amount.

alternative is remove isETH, make it totalWithDepositedAmount = totalBalance(), and do yieldToken.safeTransferFrom before _depositFor in depositYieldToken